docs/_pages/nullabletypes.md

```diff
@@ -13,12 +13,15 @@ sidebar:
 short? theShort = null;
 theShort.Should().NotHaveValue();
 theShort.Should().BeNull();
+theShort.Should().Match(x => !x.HasValue);
```

Comment:
I would prefer some examples that better reflects the use cases that `Match` is intended to cover, i.e. assertions that can't be expressed with existing ones.
E.g. `Match(x => !x.HasValue)` could be written as `NotHaveValue()`.

How about `Match(x => !x.HaveValue || x > 0)`?